pulp_container/app/registry_api.py

```diff
@@ -473,6 +477,135 @@ def get_queryset(self, *args, **kwargs):
         return (public_repositories | accessible_repositories).distinct()
 
 
+class FlatpakIndexDynamicView(APIView):
+    """
+    Handles requests to the /index/dynamic endpoint
+    """
+
+    authentication_classes = []
+    permission_classes = []
+
+    def recurse_through_manifest_lists(self, tag, manifest, oss, architectures, manifests):
+        mlms = manifest.listed_manifests.through.objects.filter(image_manifest__pk=manifest.pk)
+        if oss:
+            mlms.filter(os__in=oss)
+        if architectures:
+            mlms.filter(architecture__in=architectures)
+        for mlm in mlms:
+            self.recurse_through_manifest_lists(
+                tag, mlm.manifest_list, oss, architectures, manifests
+            )
+        if manifest.media_type in (models.MEDIA_TYPE.MANIFEST_V2, models.MEDIA_TYPE.MANIFEST_OCI):
+            manifests.setdefault(manifest, set()).add(tag)
+
+    def get(self, request):
+        req_repositories = None
+        req_tags = None
+        req_oss = None
+        req_architectures = None
+        req_label_exists = set()
+        req_label_values = {}
+        for key, values in request.query_params.lists():
+            if key == "repository":
+                req_repositories = values
+            elif key == "tag":
+                req_tags = values
+            elif key == "os":
+                req_oss = values
+            elif key == "architecture":
+                req_architectures = values
+            elif key.startswith("label:"):
+                if key.endswith(":exists"):
+                    if any(v != "1" for v in values):
+                        raise ParseError()
+                    label = key[len("label:") : len(key) - len(":exists")]
+                    req_label_exists.add(label)
+                else:
+                    label = key[len("label:") :]
+                    req_label_values[label] = values
+            else:
+                # In particularly, this covers any annotation:... parameters, which this
+                # implementation does not support:
+                raise ParseError()
```

Comment:
Is it possible to return an error message as well (https://www.django-rest-framework.org/api-guide/exceptions/#parseerror)?